src/serialization/data_file.cpp

```diff
@@ -0,0 +1,15 @@
+#include "serialization/data_file.hpp"
+#include <stdexcept>
+
+namespace found {
+
+    void DataFileHeader::validate() {
+        if (magic[0] != 'F' || magic[1] != 'O' || magic[2] != 'U' || magic[3] != 'N') {
+            throw std::runtime_error("Magic value mismatch");
+        }
```

Comment:
I think its a good thing you're completing these checks, but the choice to throw an error might be a bad one, especially since this will go onto an embedded system. I'm not asking you to change it, I'd just like to see a rationale behind why we should throw an error and not do something else (for instance, just say we have a blank data file).